internal/dispatch/graph/check_test.go

```diff
@@ -1372,6 +1373,321 @@ func TestCheckDebugging(t *testing.T) {
+				Membership: v1.ResourceCheckResult_MEMBER,
+			}),
+			false,
+		},
+		{
+			"no relationships with matching arrow check hint",
+			`definition user {}
+		
+   			 definition organization {
+				relation member: user
+			 }
+
+		 	 definition document {
+				relation org: organization
+				permission view = org->member
+  			 }`,
+			[]*core.RelationTuple{},
+			ONR("document", "somedoc", "view"),
+			ONR("user", "tom", graph.Ellipsis),
+			hints.CheckHintForArrow("document", "somedoc", "org", "member", ONR("user", "tom", graph.Ellipsis), &v1.ResourceCheckResult{
+				Membership: v1.ResourceCheckResult_MEMBER,
+			}),
+			true,
+		},
+		{
+			"no relationships with non matching tupleset arrow check hint",
+			`definition user {}
+		
+   			 definition organization {
+				relation member: user
+			 }
+
+		 	 definition document {
+				relation org: organization
+				permission view = org->member
+  			 }`,
+			[]*core.RelationTuple{},
+			ONR("document", "somedoc", "view"),
+			ONR("user", "tom", graph.Ellipsis),
+			hints.CheckHintForArrow("document", "somedoc", "anotherrel", "member", ONR("user", "tom", graph.Ellipsis), &v1.ResourceCheckResult{
+				Membership: v1.ResourceCheckResult_MEMBER,
+			}),
+			false,
+		},
+		{
+			"no relationships with non matching computed userset arrow check hint",
+			`definition user {}
+		
+   			 definition organization {
+				relation member: user
+			 }
+
+		 	 definition document {
+				relation org: organization
+				permission view = org->member
+  			 }`,
+			[]*core.RelationTuple{},
+			ONR("document", "somedoc", "view"),
+			ONR("user", "tom", graph.Ellipsis),
+			hints.CheckHintForArrow("document", "somedoc", "org", "membersssss", ONR("user", "tom", graph.Ellipsis), &v1.ResourceCheckResult{
+				Membership: v1.ResourceCheckResult_MEMBER,
+			}),
+			false,
+		},
+		{
+			"check hint over part of an intersection but missing other branch in rels",
+			`definition user {}
+
+			definition document {
+				relation editor: user
+				relation viewer: user
+				permission view = viewer & editor
+			}`,
+			[]*core.RelationTuple{},
+			ONR("document", "somedoc", "view"),
+			ONR("user", "tom", graph.Ellipsis),
+			hints.CheckHintForComputedUserset("document", "somedoc", "viewer", ONR("user", "tom", graph.Ellipsis), &v1.ResourceCheckResult{
+				Membership: v1.ResourceCheckResult_MEMBER,
+			}),
+			false,
+		},
+		{
+			"check hint over part of an intersection with other branch in rels",
+			`definition user {}
+
+			definition document {
+				relation editor: user
+				relation viewer: user
+				permission view = viewer & editor
+			}`,
+			[]*core.RelationTuple{
+				tuple.MustParse("document:somedoc#editor@user:tom"),
+			},
+			ONR("document", "somedoc", "view"),
+			ONR("user", "tom", graph.Ellipsis),
+			hints.CheckHintForComputedUserset("document", "somedoc", "viewer", ONR("user", "tom", graph.Ellipsis), &v1.ResourceCheckResult{
+				Membership: v1.ResourceCheckResult_MEMBER,
+			}),
+			true,
```

Comment:
can you add the following tests:
- hints over an exclusion
- a hint is provided per branch of an intersection
- a hint is provided per branch of an exclusion
- test the various permutations with wildcard types

Reply:
Added exclusion and wildcard tests. We don't need permutations of wildcards as they are treated the same as any other relation